rasgotransforms/rasgotransforms/transforms/aggregate/aggregate.sql

```diff
@@ -4,10 +4,14 @@ SELECT
 {%- endfor -%}
 
 {%- for col, aggs in aggregations.items() %}
-        {%- set outer_loop = loop -%}
+    {%- set outer_loop = loop -%}
     {%- for agg in aggs %}
-    {{ agg }}({{ col }}{{ ", '|'" if agg|lower == 'listagg' else ""}}) as {{ col + '_' + agg }}{{ '' if loop.last and outer_loop.last else ',' }}
+        {%- if ' DISTINCT' in agg|upper %}
+            {{ agg|replace(" DISTINCT", "") }}(DISTINCT {{ col }}) as {{ col ~ '_' ~ agg|replace(" DISTINCT", "") ~ 'DISTINCT'}}{{ '' if loop.last and outer_loop.last else ',' }}
```

Comment:
```suggestion
            {{ agg|upper|replace(" DISTINCT", "") }}(DISTINCT {{ col }}) as {{ col ~ '_' ~ agg|upper|replace(" DISTINCT", "") ~ 'DISTINCT'}}{{ '' if loop.last and outer_loop.last else ',' }}
```

Do we check anywhere that the methods passed in are uppercase? This might be unecessary if so.